### ISSUE TITLE
[REQ] typescript-axios: make prepare script configurable

### ISSUE THREAD

**Issue by forresthopkinsa** [OpenAPITools/openapi-generator#12310]
### Is your feature request related to a problem? Please describe.

Right now, the typescript-axios generator (and possibly the other TS generarors) creates a package.json with scripts like so:

```json
"scripts": {
  "build": "tsc --outDir dist/",
  "prepare": "npm run build"
},
```

`prepare` is a "lifecycle script" for NPM and Yarn, which gets run whenever the package's dependencies are installed. Clearly, Typescript transpilation/building is not part of the dependency installation step, but I can understand why this was added: it's a very helpful convenience when consuming the module with NPM's git resolution, so you don't have to compile the generated sources yourself.

Nonetheless, in more general scenarios, this behavior is incorrect and likely unwanted. The problem can be worked around on the user side with the flag `--ignore-scripts`, so it's not really a huge issue. Still, it would be nice to have the option to disable it.

## Describe the solution you'd like

A generator-specific configuration option that can preclude adding the `prepare` script

## Describe alternatives you've considered

As mentioned above, `--ignore-scripts` is a tolerable workaround. Still, it would be nice not to need it.

**Comment by michaelp** [comment]
@forresthopkinsa, good point! Please also notice that the generated section for scripts requires a global installation of typescript.  See the snippet below. As a workaround, I constantly "fix" this section by introducing `npx` (see the second snippet)
Making this section configurable would be the most flexible solution.
* Default generator
```js
"scripts": {
    "build": "tsc && tsc -p tsconfig.esm.json",
    "prepare": "npm run build"
  }
``` 
* My workaround for the `TSC is not found` problem 
```js
"scripts": {
    "build": "npx tsc && npx tsc -p tsconfig.esm.json",
    "prepare": "npm run build"
  }
```

**Comment by forresthopkinsa** [comment]
@michaelp Hmmm, this is not the expected behavior for NPM scripts. NPM adds `node_modules/.bin` to the $PATH of the environment those scripts get executed in.

**Comment by michaelp** [comment]
@forresthopkinsa, thanks for the helpful hint. I found the root cause, which differs from what I assumed in my message.
